src/migrateContentServer.test.ts

```diff
@@ -232,4 +233,97 @@ describe("migrateContentServer", () => {
       undefined,
     );
   });
+
+  it("keeps the original item untouched, as before the whole migration when the item cannot be migrated due to an error and the `behaviorOnError` flag is set to `keep-original`.", async () => {
+    const contentServer: ContentRecord = {
+      children: {
+        ui: smallLegacyUIFolderWithInvalidWidget,
+        pivot: smallLegacyPivotFolder,
+      },
+      entry: {
+        owners: [],
+        readers: [],
+        isDirectory: true,
+        canRead: true,
+        canWrite: false,
+        lastEditor: "Freddie Mercury",
+        timestamp: 0xbeef,
+      },
+    };
+
+    const contentServerBeforeMigration = _cloneDeep(contentServer);
+
+    await migrateContentServer({
+      contentServer,
+      servers,
+      fromVersion: "4.3",
+      toVersion: "5.1",
+      keysOfWidgetPluginsToRemove: [],
+      doesReportIncludeStacks: false,
+      shouldUpdateFiltersMdx: true,
+      behaviorOnError: "keep-original",
+    });
+
+    const savedWidgetContentBeforeMigration =
+      contentServerBeforeMigration.children?.ui.children?.bookmarks.children
+        ?.content.children?.["158"].entry.content;
+
+    const savedWidgetContentAfterMigration =
+      contentServer?.children?.ui?.children?.widgets?.children?.content
+        ?.children?.["158"].entry.content;
+
+    expect(savedWidgetContentBeforeMigration).toBe(
```

Comment:
Let's add some comments here:

```suggestion
    // The widget with id "158" has an invalid widget key and its migration will therefore fail.
    // Since `behaviorOnError` is set to "keep-original", the output content still contains the state of this saved widget exactly as it was before the migration.
    expect(savedWidgetContentBeforeMigration).toBe(
```